corehq/apps/es/management/commands/elastic_sync_multiplexed.py

```diff
@@ -181,6 +182,48 @@ def delete_index(self, cname):
 
         es_manager.index_delete(older_index)
 
+    def _get_current_and_older_index_name(cls, cname):
+        """
+        Returns a tuple of current index name and older index name related to the given cname.
+        Older index refers to the source index during reindex process
+        Current index refers to the destination index
+        """
+        current_index = getattr(es_consts, f"HQ_{cname.upper()}_SECONDARY_INDEX_NAME")
+        older_index = getattr(es_consts, f"HQ_{cname.upper()}_INDEX_NAME")
+        return (current_index, older_index)
+
+    def set_checkpoints_for_new_index(self, cname):
+        """
+        Takes in an index cname and create new checkpoint for all the pillows that use the older index name
+        for that cname.
+        Can only be performed when indexes are still multiplexed and not swapped.
+        When we swap the indexes, the primary index changes which updates the checkpoint names.
+        We should stop the pillows and copy the checkpoint to the new checkpoint ids, swap the indexes
+        and then start the pillows.
+        """
+        adapter = doc_adapter_from_cname(cname)
+        if not isinstance(adapter, ElasticMultiplexAdapter):
+            raise IndexNotMultiplexedException(f"""Checkpoints can be copied on multiplexed indexes.
+                Make sure you have set ES_{cname.upper()}_INDEX_MULTIPLEXED to True """)
+
+        current_index_name, older_index_name = self._get_current_and_older_index_name(cname)
+
+        assert adapter.index_name != older_index_name, f"""Current index does not match the source index.
+        Make sure that ES_{cname.upper()}_INDEX_SWAPPED is set to False"""
```

Comment:
Assertions can be disabled with a Python command-line parameter. What do you think of using an `if`-block similar to the one above?

Reply:
Sure thing. I can do that.

Reply:
https://github.com/dimagi/commcare-hq/pull/33219/commits/f7a75b98c4fd5c0dea2a0c7fd1c7ebc9c9ba5648